src/interfaces/setup_interface.jl

```diff
@@ -0,0 +1,226 @@
+    i.e. if the returned value is `true` the assert does not trigger and nothing happens, but if the returned value is `false`, 
+    an error with a generic error message will be thrown (see  [`_is_valid_input`](@ref) for details). 
+    To customize the error message, or use a custom assertion mechanism, just add your own implementation of
+
+    ```Julia
+    _assert_valid_input(stp::YourCustomSetup,input)
+    ```
+    which should throw an [`InvalidInputError`](@ref) if the input is invalid.
+    Despite the presence of a custom `_assert_valid_input`, it is highly recommended to also implement `_is_valid_input` for `CustomSetup`, because it might be used outside of the assert function.
+
+"""
+@inline function _assert_valid_input(stp::AbstractComputationSetup, input)
+    return nothing
+end
+
+"""
+
+    _is_valid_input(stp::AbstractComputationSetup, input::Any)
+
+Interface function, which returns true if the constraints of the `input` associated with the quantity of `stp` are met.
+This function is called to validate the input of [`compute`](@ref) before calling [`_compute`](@ref).
+
+!!! note "Default implementation"
+    
+    Since no input validation is equivalent to every input being valid, this function returns `true` by default. 
+    This behavior can be overwritten if actual validation is necessary.
+
+
+An assert version of this function is given by [`_assert_valid_input`](@ref), which directly uses the output of this function.
+
+"""
+@inline function _is_valid_input(stp::AbstractComputationSetup, input)
+    try
+        _assert_valid_input(stp, input)
+    catch e
+        if isa(e, AbstractInvalidInputException)
+            return false
+        end
+        @warn "The function _assert_valid_input throws an Exception, which is not an InvalidInputError! The Exception thrown is: "
+        throw(e)
+    end
+    return true
+end
+
+"""
+    
+    function _post_processing(stp::AbstractComputationSetup, input::Any, result::Any)
+    
+Interface function, which is called in [`compute`](@ref) after [`_compute`](@ref) has been called. This function is dedicated to 
+finalize the result of a computation. 
+
+!!! note "default implementation"
+
+    Since in the case of no post processing the result of [`_compute`](@ref) is unchanged, this function returns `result` by default.
+
+"""
+@inline function _post_processing(stp::AbstractComputationSetup, input, result)
+    return result
+end
+
+"""
+    
+    _compute(stp::AbstractComputationSetup, input::Any)
+
+Interface function that returns the value of the associated quantity evaluated on `input`, which can be anything the associated quantity is defined to be feasible for.
+
+!!! note "unsafe implementation"
+
+    This function must be implemented for any subtype of [`AbstractComputationSetup`]. It should not do any input validation or post processing (see [`_is_valid_input`](@ref) and [`_post_processing`](@ref)), as those two are performed while calling 
```

Comment:
Should this reference `_assert_valid_input` instead of `_is_valid_input`, since that is the one the user of the interface should implement (first)?
```suggestion
    This function must be implemented for any subtype of [`AbstractComputationSetup`]. It should not do any input validation or post processing (see [`_assert_valid_input`](@ref) and [`_post_processing`](@ref)), as those two are performed while calling 
```